### PR TITLE
My changes

### PR DESCRIPTION
**Pull request by metagn**

### CODE REVIEW

**Comment by timotheecour** [comment]
excellent changes, thanks!
I'm curious, I seem to be able to use `finally:` and in place of `defer:` ; not sure what difference it would make

File: README.md [hunk 1]
```diff
@@ -179,7 +179,9 @@ See also libraries.md
 
 ## nim questions (answered)
 * are there real immutable variables in nim ?
-A:https://www.reddit.com/r/nim/comments/2w32oi/immutability_in_nim/
+A: https://www.reddit.com/r/nim/comments/2w32oi/immutability_in_nim/
+* how to specify immutable inside `for(foo in bar)` ?
+A: for foo in bar: let foo = foo
```
Comment (ghost) [inline README.md:184]: AFAIK this is not right - immutability depends on iterator - it can return mutable or immutable value

Comment (ghost) [reply README.md:184]:
```nim
var a = @[1, 2, 3, 4]

for item in a:
  item += 5  # compile-time error

for item in mitems(a):
  item += 5  # works fine
```

Comment (ghost) [reply README.md:184]: And by default (well, it's done like this almost everywhere) iterators return immutable values

Comment (metagn) [reply README.md:184]: I was thinking about replicating D's `foreach (immutable foo; bar)` construct, and `immutable foo` seems to be the same as `let foo` in nim. Beyond that, if you don't know if foo is a var type, `let`ting it beforehand is one way to make sure. After all the question wasnt whether or not you should specify immutable or not, its asking how to do it  If my answer is still wrong though I'd love to know

**Comment by metagn** [comment]
I couldn't find documentation about finally: without try:, and since the forum post linked was in 2013 i thought it was an outdated feature in contrast to defer: which is explained using try/finally in the manual

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md:184]: @hlaaftana well, why should we replicate constructs from different languages 1 to 1? It's generally known that all iterators by default return immutable values

**Comment by timotheecour** [comment]
@hlaaftana I added 2 scope guard entries: scope(success) and scope(failure), do you know the nim equivalents?

EDIT: /cc @hlaaftana 

> I couldn't find documentation about finally: without try:

indeed, it gives:
` Warning: use 'defer'; standalone 'finally' is deprecated [Deprecated]`

File: README.md (hunk 1 above)
Comment (timotheecour) [reply README.md:184]: ok I updated that answer in 79695261b1829408b4a3f92de1bd02946b8a06d4

**Comment by metagn** [comment]
I don't know if there are any

**Pull request merged**